### PR TITLE
Use the officially supported apache mina sshd in m2k-func

### PR DESCRIPTION
**Pull request by rgolangh**
The various git command we use in m2k func now uses the officially
supported ssh implementation for the transport.

Added tests to see the transport is working well with ed25519 keys

Notice - the StrictHost option is now set using a standard ssh_config
(see man ssh_config) file which is assumed to be where the key is located -
so now we can just drop the id_rsa file in a well known location along
with the rest of the .ssh files:

/etc/.ssh/config
/etc/.ssh/known_hosts
/etc/.ssh/id_rsa

Signed-off-by: Roy Golan <rgolan@redhat.com>

### CODE REVIEW

**Comment by gabriel-farache** [comment]
@rgolangh Please also update the manifest related to [m2k-kfunc and make sure the newly needed file is also added in the initContainer](https://github.com/parodos-dev/serverless-workflows-config/blob/main/charts/move2kube/templates/03-knative-resources.yaml#L45) and update the [install readme](https://github.com/parodos-dev/serverless-workflows-config/blob/main/charts/move2kube/INSTALL.md) accordingly

**Comment by rgolangh** [comment]
> @rgolangh Please also update the manifest related to [m2k-kfunc and make sure the newly needed file is also added in the initContainer](https://github.com/parodos-dev/serverless-workflows-config/blob/main/charts/move2kube/templates/03-knative-resources.yaml#L45) and update the [install readme](https://github.com/parodos-dev/serverless-workflows-config/blob/main/charts/move2kube/INSTALL.md) accordingly

ack , I'll also update the e2e resource in this repo

File: move2kube/m2k-func/src/main/java/dev/parodos/service/GitServiceImpl.java [hunk 1]
```diff
@@ -97,27 +96,18 @@ public static TransportConfigCallback getTransport(Path sshKeyPath) throws IOExc
       throw new IOException("SSH key file at '%s' does not exists".formatted(sshKeyPath.toString()));
     }
 
-    var sshSessionFactory = new JschConfigSessionFactory() {
-      @Override
-      protected void configure(OpenSshConfig.Host host, Session session) {
-        session.setConfig("StrictHostKeyChecking", "no");
-        session.setConfig("PreferredAuthentications", "publickey");
-      }
+    var sshSessionFactory = new SshdSessionFactoryBuilder()
```
Comment (rgolangh) [inline move2kube/m2k-func/src/main/java/dev/parodos/service/GitServiceImpl.java:99]: I think that there's a good chance we don't need a custom session factory now that we can you proper .ssh config folder. If we mount the key in the supported defalt name , say id_rsa or id_ed25519 it should work out of the box. Alternatively we could say in the ssh config `IdentityFile /path/to/key`

Comment (rgolangh) [reply move2kube/m2k-func/src/main/java/dev/parodos/service/GitServiceImpl.java:99]: and of course make publickey the preferred auth
```
Host *
    PreferredAuthentications publickey
```

File: e2e/resources/knative-service.yaml [hunk 2]
```diff
@@ -1,11 +1,11 @@
 apiVersion: v1
-data:
-  config: |
-    Host *
-      StrictHostKeyChecking no
 kind: ConfigMap
 metadata:
   name: m2k-ssh-config
+data:
```
Comment (masayag) [inline e2e/resources/knative-service.yaml:5]: The ci complains for this:
```
Error from server (BadRequest): error when creating "e2e/resources/knative-service.yaml": ConfigMap in version "v1" cannot be handled as a ConfigMap: strict decoding error: unknown field "spec"
service.serving.knative.dev/m2k-save-transformation-func created
```

Comment (masayag) [reply e2e/resources/knative-service.yaml:5]: which is pretty funny since there isn't a real change from the previous version (only location was changed)

**Comment by rgolangh** [comment]
@gabriel-farache Fixed the e2e and now they run on the tip of of the PR
There is a race between the deployment of the m2k workflow and the e2e tests because there's no wait to see the workflow is visible from backstage, probably because it takes time till the data-index registers it. We should address that in another PR

**Review by masayag — APPROVED** [review]

**Pull request merged**